crates/oxc_semantic/src/builder.rs

```diff
@@ -203,10 +206,23 @@ impl<'a> SemanticBuilder<'a> {
     }
 
     fn create_ast_node(&mut self, kind: AstKind<'a>) {
+        let span = kind.span();
+
+        let mut leading_comments = vec![];
+        // Should handle trailing comments as well?
+        for (start, comment) in self.trivias.comments().range(..span.start) {
```

Comment:
This will hurt performance I think, because it's run on every ast node :-(

Reply:
Is it OK if I limit the target kinds like original code does? (although need to find out what to target from TS codebase ⛰️  )

Reply:
Yes it's ok